main.f90

```diff
@@ -1003,3 +1005,23 @@ subroutine calcfr(pp,pair,sum4)
   enddo
   
 end subroutine calcfr
+
+
+! Get the absolute path to file
+! Note, this function replaces shell tilde ~/ with the explicit home dir string.
+function fullpath(fname)
+  implicit none
+  character fname*(*)
+  character(len=4096) :: absdir, fullpath
+  if(fname(1:1) .eq. '/') then
+    fullpath = fname
+  else if(fname(1:2) .eq. '~/') then
+    call getenv('HOME', absdir)
+    fullpath = absdir(:lnblnk(absdir)) // fname(2:lnblnk(fname))
```

Comment:
`lnblnk` is a GNU extension, please use `len_trim` instead (see https://gcc.gnu.org/onlinedocs/gfortran/LNBLNK.html)

Reply:
Ok, no problem, I've already forgotten Fortran...